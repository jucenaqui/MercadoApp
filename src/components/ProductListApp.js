import React from 'react'

export const ProductListApp = () => {
    return (
        <div>
            <h1>Product list</h1>
        </div>
    )
}
