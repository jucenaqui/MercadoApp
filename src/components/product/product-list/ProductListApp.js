import React, { useEffect, useState } from 'react';
import { useParams } from "react-router-dom";

import { getProductsById } from '../../../hooks/GetProductsById';
import { ProductItemDescriptionApp } from '../producto-item-description/ProductItemDescriptionApp';

export const ProductListApp = () => {

    const { id } = useParams();

    const [productItem, setProductItem] = useState({})
    
    const item = productItem;

    useEffect(() => {
        getProductsById( id )
        .then( prods => {
           setProductItem( prods );
           console.log(prods)
        })
    }, [id]);

    return (
        <div>
            {
                item && <ProductItemDescriptionApp product={item} />
            }
        </div>
    )
}
