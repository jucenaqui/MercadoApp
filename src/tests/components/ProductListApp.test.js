import { shallow } from "enzyme";
import React from 'react'
import ReactRouter from 'react-router';


import { ProductListApp } from "../../components/product/product-list/ProductListApp";

describe('Test to <ProductListApp />', () => {
    

    jest.spyOn(ReactRouter, 'useParams').mockReturnValue({ id: 'MLA856481792' });
    const wrapper = shallow(<ProductListApp />);
   

    test('should show the component succesfull', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should have a field categories array', () => { 
        const productIDes = wrapper.find('ProductItemDescriptionApp');
        expect( Array.isArray(productIDes.prop('categories'))).toBe(true);
    });

    test('should have a field product object', () => { 
        const productIDes = wrapper.find('ProductItemDescriptionApp');
        expect(typeof productIDes.prop('categories')).toBe('object');
    });

})