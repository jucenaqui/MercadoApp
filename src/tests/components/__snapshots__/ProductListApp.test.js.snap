// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to <ProductListApp /> should show the component succesfull 1`] = `
<div>
  <ProductItemDescriptionApp
    categories={Array []}
    product={Object {}}
  />
</div>
`;
